### PR TITLE
feat(sampling): alpha fastmax mode

### PR DESCRIPTION
**Pull request by yakkomajuri**
Moving fast here! Remember sampling is still in Alpha and behind a feature flag. Very keen to get this live and have a couple people test it out.

This is a pretty cool feature that allows you to turn on "fast mode" for the current session. It's a purely frontend thing and refreshing the page manually will get rid of it. What it does is set 10% sampling on all insights you refresh after enabling it, making everything much faster.

A few caveats with this Alpha version that will be covered later:

- UI design: we won't be using a lightning bolt, but rather a Max running or something like that. Other aspects of the UI are completely changeable too
- Docs/discoverability: we need to make things more clear as to how this works (e.g. insight filters override this - so if you set a sampling rate of 1% on the insight it will override fast mode's 10%)
- Turning this on means you actually set filters on an insight. i.e. if you have this on and save an insight, coming back later in non-fast mode will still have it saved as sampled. Something for us to think on, but it _should_ be fine given we're working on making it very clear to users that the insight is sampled or not, both on dashboards and insight pages
- We may later want to actually store user preferences in the db

Screenshot 2023-03-02 at 15 05 02
Screenshot 2023-03-02 at 15 05 16

### CODE REVIEW

File: frontend/src/scenes/userLogic.ts [hunk 1]
```diff
@@ -89,6 +91,12 @@ export const userLogic = kea<userLogicType>([
                 }),
             },
         ],
+        globalSessionFilters: [
```
Comment (pauldambra) [inline frontend/src/scenes/userLogic.ts:94]: 🤔 longer term I think this lives better off the user. insight logics could store this into a shared location in storage and read/clear that as appropriate.

Or it becomes a team setting... "turn this on for all" like with local account/test filters

But no reason to block this here!

**Review by pauldambra — APPROVED** [review]
Screenshot 2023-03-03 at 12 36 35

Not blocking for this but query id has ended up off to one side in the slow loading view

File: frontend/src/scenes/userLogic.ts (hunk 1 above)
Comment (yakkomajuri) [reply frontend/src/scenes/userLogic.ts:94]: yeah wanted to have it on insight logic at first but need something that persists across the session. Local/session storage could be the way to go

**Comment by yakkomajuri** [comment]
thanks for the heads up on the query id - fixed

**Pull request merged**